### PR TITLE
RST-382 Fixing jitter

### PR DESCRIPTION
**Pull request by ayrton04**
Addresses https://locusrobotics.atlassian.net/browse/RST-382

Goes with https://github.com/locusrobotics/locus_bots/pull/138

@locusrobotics/robot-software-team

### CODE REVIEW

File: src/ros_filter.cpp [hunk 1]
```diff
@@ -1754,8 +1766,17 @@ namespace RobotLocalization
...
-                           odomBaseLinkTrans);
+              if (!RosFilterUtilities::lookupTransformSafe(
+                     tfBuffer_,
+                     baseLinkFrameId_,
+                     odomFrameId_,
+                     ros::Time(filter_.getLastMeasurementTime()),
+                     odomBaseLinkTrans,
+                     true))
+              {
+                ROS_ERROR_STREAM("Unable to retrieve " << odomFrameId_ << "->" << baseLinkFrameId_ << " transform. Skipping iteration...");
```
Comment (tappan-at-git) [inline src/ros_filter.cpp:1777]: Should this include the last measurement time? If it differs from the log timestamp it could be significant.

**Review by tappan-at-git — APPROVED** [review]

File: src/ros_filter.cpp [hunk 2]
```diff
@@ -549,28 +551,35 @@ namespace RobotLocalization
...
 
+    if (filter_.getInitializedStatus() && predictToCurrentTime)
+    {
+      double lastUpdateDelta = currentTimeSec - filter_.getLastMeasurementTime();
+
+      filter_.validateDelta(lastUpdateDelta);
+      filter_.predict(currentTimeSec, lastUpdateDelta);
+
+      // Update the last measurement time and last update time
+      filter_.setLastMeasurementTime(filter_.getLastMeasurementTime() + lastUpdateDelta);
```
Comment (svwilliams) [inline src/ros_filter.cpp:579]: I do not fully understand how MeasurementTime and UpdateTime are being used in the rest of the code, but the names would suggest that MeasurementTime is coming from sensor data. Here you are updating the MeasurementTime to currentTimeSec even though no measurements have been received at currentTimeSec. Setting the UpdateTime makes sense to me.

**Review by svwilliams — COMMENTED** [review]
One question about updating the measurement time.

File: src/ros_filter.cpp (hunk 2 above)
Comment (ayrton04) [reply src/ros_filter.cpp:579]: Soooo long story short: `updateTime` needs to go. That, or I need to just remove `lastMeasurementTime` as a variable and rename everywhere it is used to `lastUpdateTime`. I originally wrote it to handle systems in which the measurement data was out-of-sync time-wise with the EKF executable, but that's stupid: the sensors are going to be onboard and should be based on the same clock. Moreover, I think the onus should be on the user to ensure clocks are synced.

Comment (ayrton04) [reply src/ros_filter.cpp:579]: I can fix all of that as part of this PR.

File: src/ros_filter.cpp (hunk 1 above)
Comment (ayrton04) [reply src/ros_filter.cpp:1777]: Sure

**Review by svwilliams — APPROVED** [review]

File: src/ros_filter.cpp (hunk 2 above)
Comment (svwilliams) [reply src/ros_filter.cpp:579]: As long as it was intentional, that's fine.

**Pull request merged**